v0/destinations/vero/transform.js

```diff
@@ -0,0 +1,211 @@
+const commonResponseBuilder = (message, category, destination) => {
+  const payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  const { authToken } = destination.Config;
+  set(payload, "auth_token", authToken);
+
+  // channels is defined only for Identify call
+  if (category.type === EventType.IDENTIFY) {
+    const address = get(payload, "channels.address");
+    const platform = get(payload, "channels.platform");
+
+    // Vero only accepts a valid channels object - address, platform and type should be defined and not null
+    // Added this block because channels & channels.device can be defined and not null, which should be removed
+    if (!(isDefinedAndNotNull(address) && isDefinedAndNotNull(platform))) {
+      delete payload.channels;
+    }
+  }
+
+  // Setting required tag fields
+  if (category.type === "tags") {
+    const integrationsObj = getIntegrationsObj(message, "vero");
+    const addTags = get(integrationsObj, "tags.add");
+    const removeTags = get(integrationsObj, "tags.remove");
+    set(payload, "add", addTags);
+    set(payload, "remove", removeTags);
+  }
+
+  const response = defaultRequestConfig();
+  response.endpoint = category.endpoint;
+  response.body.JSON = removeUndefinedAndNullValues(payload);
+  response.method = category.method;
+
+  return response;
+};
+
+// This block handles any tag addition or removal requests.
+// Ref - https://developers.getvero.com/?bash#tags
+const tagResponseBuilder = (message, category, destination) => {
+  const integrationsObj = getIntegrationsObj(message, "vero");
+  const addTags = get(integrationsObj, "tags.add");
+  const removeTags = get(integrationsObj, "tags.remove");
+  let tagResponse = {};
+  // Both add and remove should be an array and at least one should have a length > 0
+  if (
+    (Array.isArray(removeTags) && removeTags.length > 0) ||
+    (Array.isArray(addTags) && addTags.length > 0)
+  ) {
+    const tagCategory = CONFIG_CATEGORIES.TAGS;
+    tagResponse = commonResponseBuilder(message, tagCategory, destination);
+  }
+
+  return tagResponse;
+};
+
+// This function handles identify requests as well as addition-removal of tags.
+// Identify Ref - https://developers.getvero.com/?bash#users-identify
+const identifyResponseBuilder = (message, category, destination) => {
+  const respList = [];
+  const response = commonResponseBuilder(message, category, destination);
+  respList.push(response);
+
+  // Handling tags
+  const tagResponse = tagResponseBuilder(message, category, destination);
+  if (isDefinedAndNotNullAndNotEmpty(tagResponse)) {
+    respList.push(tagResponse);
+  }
+
+  return respList;
+};
+
+// This function handles track requests as well as addition-removal of tags.
+// Track Ref - https://developers.getvero.com/?bash#events-track
+const trackResponseBuilder = (message, category, destination) => {
+  if (
+    message.event.toLowerCase() === "unsubscribe" ||
+    message.event.toLowerCase() === "resubscribe"
```

Comment:
While matching we should always match using const (string stored in const varianbles this prevents vulnurability)

Reply:
Done